File: e2e/java/src/test/java/dev/kreuzberg/e2e/StealthTest.java
package dev.kreuzberg.e2e;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;
import dev.kreuzberg.kreuzcrawl.Kreuzcrawl;

/** E2e tests for category: stealth. */
class StealthTest {
    @Test
    void testStealthUaRotationConfig() throws Exception {
        // User-agent rotation config is accepted and crawl succeeds
        var engine = Kreuzcrawl.createEngine(null);
        String url = System.getenv("MOCK_SERVER_URL") + "/fixtures/stealth_ua_rotation_config";
        var result = Kreuzcrawl.scrape(engine, url);
        assertEquals(200, result.statusCode());
    }

}

File: e2e/elixir/test/stealth_test.exs
# E2e tests for category: stealth
defmodule E2e.StealthTest do
  use ExUnit.Case, async: true

  describe "stealth_ua_rotation_config" do
    test "User-agent rotation config is accepted and crawl succeeds" do
      engine = Kreuzcrawl.create_engine!(nil)
      url = System.get_env("MOCK_SERVER_URL") <> "/fixtures/stealth_ua_rotation_config"
      result = Kreuzcrawl.scrape!(engine, url)
      assert String.trim(result.status_code) == 200
    end
  end
end
